src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -577,13 +603,16 @@ void ExpressionAnalyzer::getRootActions(const ASTPtr & ast, bool no_makeset_for_
         settings.size_limits_for_set,
         subquery_depth,
         sourceColumns(),
+        aggregation_keys,
+        aggregation_keys_indexes_list,
```

Comment:
Maybe we can put `aggregation_keys`, `aggregation_keys_indexes_list` and `group_by_kind` into a single structure.
And, please, add a comment what every field means.